packages/admin/admin/src/form/FinalFormNumberInput.tsx

```diff
@@ -0,0 +1,80 @@
+import { InputBase, InputBaseProps } from "@mui/material";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+import { useIntl } from "react-intl";
+
+import { ClearInputAdornment } from "../common/ClearInputAdornment";
+
+export type FinalFormNumberInputProps = InputBaseProps &
+    FieldRenderProps<number | string> & {
+        clearable?: boolean;
+        decimals?: number;
+    };
+
+export function FinalFormNumberInput({
+    meta,
+    input,
+    innerRef,
+    clearable,
+    endAdornment,
+    decimals = 0,
+    ...props
+}: FinalFormNumberInputProps): React.ReactElement {
+    const intl = useIntl();
+
+    const numberParts = intl.formatNumberToParts(1111.111);
+    const decimalSymbol = numberParts.find(({ type }) => type === "decimal")?.value;
+    const thousandSeparatorSymbol = numberParts.find(({ type }) => type === "group")?.value;
+
+    const [formattedNumberValue, setFormattedNumberValue] = React.useState("");
+
+    const getFormattedValue = React.useCallback(
+        (value, decimals: number | undefined) => {
+            const formattedValue =
+                value !== undefined ? intl.formatNumber(value, { minimumFractionDigits: decimals, maximumFractionDigits: decimals }) : "";
+            return formattedValue;
+        },
+        [intl],
+    );
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+        setFormattedNumberValue(value);
+    };
+
+    const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+
+        const numericValue = parseFloat(value.split(`${thousandSeparatorSymbol}`).join("").split(`${decimalSymbol}`).join("."));
```

Comment:
You don't need template strings here when using the variables directly

```suggestion
        const numericValue = parseFloat(value.split(thousandSeparatorSymbol).join("").split(decimalSymbol).join("."));
```